### PR TITLE
Add support for comment update to github comment task

### PR DESCRIPTION
**Pull request by afrittoli**
# Changes

Enhance the GitHub add comment task with the ability to update
a comment if applicable. This achieved by tagging comments with
and invisible (HTML comment) unique text, and looking for that
text to identify a comment.

This does not use the nodeid because that is generated by GitHub
and it would require task users to store known node IDs somewhere
across task executions.

Extend to test logic to be able to load the fixture reponse from
a file, so that we don't have to inline a long JSON doc into
the fixture config YAML. This is achieved by creating a config map
with all the files in the fixtures folder, and mounting that
configmap as a volume to the sidecar.

Signed-off-by: Andrea Frittoli <andrea.frittoli@gmail.com>

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality of task changed or new task added)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [x] Complies with [Catalog Organization TEP][TEP], see [example]. **Note** [An issue has been filed to automate this validation][validation]
  - [x] File path follows  `<kind>/<name>/<version>/name.yaml`
  - [x] Has `README.md` at `<kind>/<name>/<version>/README.md`
  - [x] Has mandatory `metadata.labels` - `app.kubernetes.io/version` the same as the `<version>` of the resource
  - [x] Has mandatory `metadata.annotations` `tekton.dev/pipelines.minVersion`
  - [x] mandatory `spec.description` follows the convention

          ```

          spec:
            description: >-
              one line summary of the resource

              Paragraph(s) to describe the resource.
          ```

_See [the contribution guide](https://github.com/tektoncd/catalog/blob/master/CONTRIBUTING.md)
for more details._

---

[TEP]: https://github.com/tektoncd/community/blob/master/teps/0003-tekton-catalog-organization.md
[example]: https://github.com/tektoncd/catalog/tree/master/task/git-clone/0.1
[validation]:  https://github.com/tektoncd/catalog/issues/413

Partially fixes: https://github.com/tektoncd/website/issues/252

### CODE REVIEW

**Comment by afrittoli** [comment]
This still needs some testing. Tests needs to be written for the "update" case.

File: task/github-add-comment/0.4/README.md [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+  (Use Case can be after running the CI you want to output the logs as
+  a part of the CI.)
+- Parameter `COMMENT` has been changed to `COMMENT_OR_FILE` in which user can
+  provide the filename in case comments are present inside some file or you can
+  directly pass the comments as well.
+
+## Install the Task
+
+```
+kubectl apply -f https://raw.githubusercontent.com/tektoncd/catalog/main/task/github-add-comment/0.3/github-add-comment.yaml
```
Comment (vinamra28) [inline task/github-add-comment/0.4/README.md:19]:
```suggestion
kubectl apply -f https://raw.githubusercontent.com/tektoncd/catalog/main/task/github-add-comment/0.4/github-add-comment.yaml
```

File: task/github-add-comment/0.4/README.md [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+
+## Changelog
+
+- `Optional Workspace` support is added so that if you want to pass a file
+  which contains the comment then it can be passed via workspace.
+  (Use Case can be after running the CI you want to output the logs as
+  a part of the CI.)
+- Parameter `COMMENT` has been changed to `COMMENT_OR_FILE` in which user can
+  provide the filename in case comments are present inside some file or you can
+  directly pass the comments as well.
```
Comment (vinamra28) [inline task/github-add-comment/0.4/README.md:14]: I guess changelog can be updated. WDYT?

File: task/github-add-comment/0.4/github-add-comment.yaml [hunk 3]
```diff
@@ -0,0 +1,177 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: github-add-comment
+  labels:
+    app.kubernetes.io/version: "0.3"
```
Comment (vinamra28) [inline task/github-add-comment/0.4/github-add-comment.yaml:7]:
```suggestion
    app.kubernetes.io/version: "0.4"
```

**Comment by chmouel** [comment]
The diff 

https://paste.gg/p/anonymous/e16da4e2c7c74e6f8c68955b2218dfeb

File: task/github-add-comment/0.4/README.md (hunk 1 above)
Comment (afrittoli) [reply task/github-add-comment/0.4/README.md:19]: fixed

File: task/github-add-comment/0.4/README.md (hunk 2 above)
Comment (afrittoli) [reply task/github-add-comment/0.4/README.md:14]: Yes, definitely - this was just a WIP 🙏

File: task/github-add-comment/0.4/github-add-comment.yaml (hunk 3 above)
Comment (afrittoli) [reply task/github-add-comment/0.4/github-add-comment.yaml]: Fixed

**Comment by afrittoli** [comment]
Diff https://gist.github.com/afrittoli/c910e88c7ce1081737b467729607ae3b

**Comment by afrittoli** [comment]
@vinamra28 @chmouel @vdemeester I addressed the comments and added tests, let me know what you think.

**Comment by chmouel** [comment]
Nice one, just a nitpick, is there any chances if you can split the commits between the task and the test? 🙏🏻

**Comment by afrittoli** [comment]
> Nice one, just a nitpick, is there any chances if you can split the commits between the task and the test? 🙏🏻

I can do. Would it be the test infra changes first or the task changes first?

**Comment by afrittoli** [comment]
@chmouel I split the PR into two commits.

**Review by vinamra28 — COMMENTED** [review]
/lgtm
/woof

**Comment by chmouel** [comment]
/lgtm
/approve

thank you

**Comment by chmouel** [comment]
I forgot to add this, we probably would want to add this new capability for fixtures to the documentation : 

https://github.com/tektoncd/catalog/blob/main/CONTRIBUTING.md#end-to-end-testing-for-external-services

**Pull request merged**